lib/initialForm/CountrySelectorWidget.dart

```diff
@@ -0,0 +1,126 @@
+import 'package:flutter/material.dart';
+import 'package:country_code_picker/country_code_picker.dart';
+import 'package:mazilon/util/styles.dart';
+import 'package:flutter_screenutil/flutter_screenutil.dart';
+import 'package:mazilon/util/userInformation.dart';
+import 'package:shared_preferences/shared_preferences.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+import 'package:provider/provider.dart';
+
+class CountrySelectorWidget extends StatefulWidget {
+  final String text;
+  final String disclaimerText;
+
+  const CountrySelectorWidget(
+      {Key? key, required this.text, required this.disclaimerText})
+      : super(key: key);
+
+  @override
+  _CountrySelectorWidgetState createState() => _CountrySelectorWidgetState();
+}
+
+class _CountrySelectorWidgetState extends State<CountrySelectorWidget> {
+  String? _selectedCountryCode = 'IL';
+  bool isvisible = false;
+  void saveLocation(String location, UserInformation userInfo) async {
+    SharedPreferences prefs = await SharedPreferences.getInstance();
+    prefs.setString("location", location);
+    userInfo.updateLocation(location);
+  }
+
+  void changeVisible() {
+    print("pressed");
+    setState(() {
+      isvisible = !isvisible;
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final userInfoProvider = Provider.of<UserInformation>(context);
+    final appLocale = AppLocalizations.of(context);
```

Comment:
We can have this as a class property and use it throughout the class
Another approach is to have it as global var 
Another (better?) approach is to derive from `StatefulWidget` and create a MazilonStatefulWidget that include the `locale` object